src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -71,9 +71,9 @@ private void AddExceptionDiagnostic(Diagnostic diagnostic)
         /// 3) Diagnostic suppression through applied <see cref="System.Diagnostics.CodeAnalysis.SuppressMessageAttribute"/>.
         /// 4) Pragma directives for the given <paramref name="compilation"/>.
         /// </summary>
-        public static IEnumerable<Diagnostic> GetEffectiveDiagnostics(IEnumerable<Diagnostic> diagnostics, Compilation compilation)
+        public static IEnumerable<Diagnostic> GetEffectiveDiagnostics(ImmutableArray<Diagnostic> diagnostics, Compilation compilation)
         {
-            if (diagnostics == null)
+            if (diagnostics == default(ImmutableArray<Diagnostic>))
             {
                 throw new ArgumentNullException(nameof(diagnostics));
             }
```

Comment:
As `ImmutableArray` is a value type I don't think the `ArgumentNullException` is appropriate or necessary. Couldn't we simply treat `default(ImmutableArray<Diagnostic>)` as though we had an empty `ImmutableArray`?